apis/Google.Cloud.Firestore/Google.Cloud.Firestore/AggregateQuery.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Cloud.Firestore.V1;
+using Google.Protobuf;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Google.Cloud.Firestore.V1.StructuredAggregationQuery.Types;
+
+namespace Google.Cloud.Firestore;
+
+/// <summary>
+/// A query for running an aggregation over a [StructuredQuery][google.firestore.v1.StructuredQuery]. Currently only "count(*)" aggregation is supported.
+/// </summary>
+public sealed class AggregateQuery : IEquatable<AggregateQuery>
+{
+    private readonly Query _query;
+
+    /// <summary>
+    /// The "alias" to specify in the <see cref="RunAggregationQueryRequest"/> proto when running a count query.
+    /// The actual value is not meaningful, but will be used to get the count out of the <see cref="RunAggregationQueryResponse"/>.
+    /// </summary>
+    private const string CountAlias = "Count";
+
+    private readonly IReadOnlyList<Aggregation> _aggregations;
+
+    internal AggregateQuery(Query query) =>
+        _query = GaxPreconditions.CheckNotNull(query, nameof(query));
+
+    private AggregateQuery(Query query, IReadOnlyList<Aggregation> aggregations)
+    {
+        _query = GaxPreconditions.CheckNotNull(query, nameof(query));
+        _aggregations = aggregations;
+    }
+
+    internal AggregateQuery WithAggregation(Aggregation aggregation)
+    {
+        if (aggregation.Count != null)

```

Comment:
Rather than splitting the initialization of the Count aggregation, why not create an internal `CreateCountAggregation` method in here, which can create it fully?